### PR TITLE
zonal_statistics: add vector_file support for "histogram" and "sd"

### PR DESCRIPTION
**Pull request by soxofaan**
internal ticket EP-3399 

- simplify and deduplicate code by using new ProjectedPolygons from geotrellis-extensions
- add tests for all aggregate_polygon funcs and polygon sources

depends on https://github.com/Open-EO/openeo-geotrellis-extensions/pull/2

### CODE REVIEW

**Comment by soxofaan** [comment]
tests currently fail because there is no official openeo-geotrellis-extensions build yet with  Open-EO/openeo-geotrellis-extensions#2

File: openeogeotrellis/GeotrellisImageCollection.py [hunk 1]
```diff
@@ -514,131 +519,102 @@ def timeseries(self, x, y, srs="EPSG:4326") -> Dict:
...
+            else:
+                raise ValueError(func)
 
             return AggregatePolygonResult(
                 timeseries=self._as_python(stats),
+                # TODO: regions can also be a string (path to vector file) instead of geometry object
                 regions=regions if multiple_geometries else GeometryCollection([regions]),
             )
-        else:  # defaults to mean, historically
+        elif func == "mean":  # defaults to mean, historically
```
Comment (bossie) [inline openeogeotrellis/GeotrellisImageCollection.py:562]: if it no longer defaults to "mean" I'd get rid of the comment

Comment (soxofaan) [reply openeogeotrellis/GeotrellisImageCollection.py:562]: True.
I don't think we have to support "default to mean" anymore

**Comment by soxofaan** [comment]
squashed and rebased on latest master

**Comment by jdries** [comment]
Looks good to me! This also answers my question in previous PR about Python side of things.

**Comment by soxofaan** [comment]
now that https://github.com/Open-EO/openeo-geotrellis-extensions/pull/2 is merged and jar has been built:
rebased and force pushed

**Comment by soxofaan** [comment]
:tada:   tests pass now

**Pull request merged**